data_visualization_in_open_source_software/data_visualization_in_open_source_software.md

```diff
@@ -0,0 +1,287 @@
+</div>
+****
+
+True or False: It's generally NOT necessary to reformat data before plotting.
+
+[( )] TRUE
+[(X)] FALSE
+****
+<div class = "answer">
+Although some data reformatting can be completed as part of the plotting code, in many cases it will be necessary (or at least a lot easier!) to do any data manipulation before plotting.
+</div>
+****
+
+List as many plot elements as you can that can be used as tools to communicate data values.
+
+[[color, position on axis/dimension, shape, line type, size, facet]]
+[[?]] Hint: Go back to [Plot elements](#plot-elements) and pay special attention to the **bold** terms.
+<script>
+  let input = "@input".trim();
+  /color|dimension|shape|line type|size|facet/i.test(input);
+</script>
+****
+<div class = "answer">
+Correct answers include color, position on axis/dimension, shape, line type, size, and facet. If you got at least a couple items from the above list, give yourself a pat on the back!
+</div>
+****
+
+Which of the following statements is the **most** true of a typical workflow for building new data visualizations?
+
+[( )] People new to data visualization should start by copying code from example plots, but as they get better they'll be able to write plot code themselves
+[( )] It's best to plan your plot carefully from the beginning so that you can get it right the first time and don't have to re-work it
+[(X)] You should pick a plot type that is a good fit both for the types of variables you have and what idea you want your plot to communicate
+[[?]] Hint: Although all of these answers have some element of truth to them, only one is **completely** true.
+[[?]] Hint: See [data visualization workflow](#data-visualization-workflow) for a review.
+****
+<div class = "answer">
+Although the first two answers seem plausible, they don't really describe a typical workflow:
+
+- Everyone, including experts, reuses code from example plots and previous visualizations (that's why [galleries of example plots](#data-visualization-workflow) are so popular!). You shouldn't expect to be able to write `ggplot2` or `seaborn` code from a blank slate. As with so many programming skills, the main difference between a `ggplot2`/`seaborn` expert and a novice is that the expert knows exactly what to google!
+- Although planning a plot carefully before you start is a good idea and will save you time, don't expect that it will save you from having to re-work it. Creating a good visualization is almost always an iterative process --- that's one reason it's such an advantage to do your plotting in a scripted language like R or python! If you want to make a small change to a plot, you don't have to start from scratch, you can just tweak the code and re-run it.
+</div>
```

Comment:
Yeah, I hear that. What I'm trying to do with this is draw their attention to some of the nuance they may have missed in the workflow description, especially the fact that even experts don't expect to be able to write perfect plot code on the first try without referencing examples etc. But I'll rethink this question completely and see if I can get at that from another angle entirely.

Reply:
I feel like I got a non-sneaky question to replace this one that still does the work I was hoping this would do re drawing learners' attention to some of the nuance of the workflow section. Let me know what you think!